util/linuxfw/nftables_runner.go

```diff
@@ -569,6 +569,12 @@ type NetfilterRunner interface {
 	// the Tailscale interface, as used in the Kubernetes egress proxies.
 	DNATNonTailscaleTraffic(exemptInterface string, dst netip.Addr) error
 
+	EnsurePortMapRuleForSvc(svc, tun string, targetIP netip.Addr, pm PortMap) error
+
+	DeletePortMapRuleForSvc(svc, tun string, targetIP netip.Addr, pm PortMap) error
+
+	DeleteSvc(svc, tun string, targetIPs []netip.Addr, pm []PortMap) error
+
```

Comment:
cc @raggi  @KevinLiang10 - just an fyi that this adds a couple more methods to the netfilter runner, these are only used for the egress proxies, no changes for the router functionality
At the moment I think there is a clean split between the iptables/nftables methods used for the router and the kube proxies, but also for future we are considering implementing the kube TCP proxy in code, so I am not giving much thought to whether we want to put this functionality somewhere else.